frontend/packages/dev-console/src/components/project-access/ProjectAccessForm.tsx

```diff
@@ -0,0 +1,198 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Formik } from 'formik';
+import { k8sCreate, k8sKill, k8sPatch } from '@console/internal/module/k8s';
+import { RoleBindingModel } from '@console/internal/models';
+import { LoadingBox, PageHeading } from '@console/internal/components/utils';
+import { generateSecret } from '../import/import-submit-utils';
+import { withStartGuide } from '../../../../../public/components/start-guide';
+import {
+  validationSchema,
+  UserRole,
+  UserRoleBinding,
+  getRolesWithNameChange,
+  getNewRoles,
+  getRemovedRoles,
+  getfinalRoles,
+} from './projectAccessForm-utils';
+
+export interface ProjectAccessFormProps {
+  ProjectAccessFormComponent: React.ComponentType<any>;
+  formName: string;
+  namespace: string;
+  roleBindings?: { data: []; loaded: boolean };
+  projects?: { data: []; loaded: boolean };
+}
+
+const ProjectAccessForm: React.FC<ProjectAccessFormProps> = ({
+  ProjectAccessFormComponent,
+  formName,
+  namespace,
+  roleBindings,
+  projects,
+}) => {
+  if (!projects.loaded || (projects.loaded && !_.isEmpty(projects.data) && !roleBindings.loaded)) {
+    return <LoadingBox />;
+  }
+
+  const noProjectsAvailable = projects.loaded && _.isEmpty(projects.data);
```

Comment:
I don't understand why projects are being fetched for this form. We are only interested in rolebindings as the for is only displayed within the scope of a project.

Reply:
@christianvogt I did this because the role Binding request to fetch all the roleBindings, when no projects exist, fails with response as forbidden and the value of `roleBindings.loaded` never becomes `true`. I thought if I simply check for `roleBindings.loaded` it would be difficult to differentiate between scenarios like `when no projects exist` & `user not having permission to list roleBindings`

Reply:
Firehose also returns errors that you can inspect.
If you set `optional` to `true` for the resource fetch, you will get no results even the user has no permissions.
Furthermore you can use rbac to enable the form if they have permission to create role bindings.

Reply:
cc @invincibleJai and @vikram-raj who have worked in these areas.

Reply:
@invincibleJai worked on adding ```optional: true``` for knative resource in the firehouse.
https://github.com/openshift/console/pull/2905

Reply:
- added `optional: true` for the fetching the resources
- added `loadError` to inspect firehose errors